### ISSUE TITLE
sample.drivers.flash.shell fails to build for adafruit_kb2040

### ISSUE THREAD

**Issue by stephanosio** [zephyrproject-rtos/zephyr#54557]
**Describe the bug**
`samples/drivers/flash_shell/sample.drivers.flash.shell` fails to build for `adafruit_kb2040`.

**To Reproduce**
```
scripts/twister -v -N -M -p adafruit_kb2040 -s samples/drivers/flash_shell/sample.drivers.flash.shell
```

**Expected behavior**
Test builds successfully.

**Impact**
Unclear

**Logs and console output**
```
Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/samples/drivers/flash_shell
-- CMake version: 3.20.5
-- Found Python3: /usr/bin/python3.8 (found suitable exact version "3.8.10") found components: Interpreter 
-- Cache files will be written to: /__w/zephyr/zephyr/.cache
-- Zephyr version: 3.3.0-rc1 (/__w/zephyr/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: adafruit_kb2040
-- Found host-tools: zephyr 0.15.2 (/opt/toolchains/zephyr-sdk-0.15.2)
-- Found toolchain: zephyr 0.15.2 (/opt/toolchains/zephyr-sdk-0.15.2)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.15.2/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/adafruit_kb2040/adafruit_kb2040.dts
-- Generated zephyr.dts: /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/dts.cmake
/__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/zephyr.dts:67.34-72.5: Warning (simple_bus_reg): /soc/peripheral-clk: missing or empty reg/ranges property
/__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/zephyr.dts:73.26-78.5: Warning (simple_bus_reg): /soc/system-clk: missing or empty reg/ranges property
Parsing /__w/zephyr/zephyr/samples/drivers/flash_shell/Kconfig
Loaded configuration '/__w/zephyr/zephyr/boards/arm/adafruit_kb2040/adafruit_kb2040_defconfig'
Merged configuration '/__w/zephyr/zephyr/samples/drivers/flash_shell/prj.conf'
Configuration saved to '/__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/.config'
Kconfig header saved to '/__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/include/generated/autoconf.h'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    TC_RUNID


-- Build files have been written to: /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell
[1/186] Preparing syscall dependency handling

[2/186] Generating include/generated/version.h
-- Zephyr version: 3.3.0-rc1 (/__w/zephyr/zephyr), build: v3.3.0-rc1-97-g432ff20a72e1
[3/186] Generating misc/generated/syscalls_subdirs.trigger
[4/186] Creating directories for 'second_stage_bootloader'
[5/186] No download step for 'second_stage_bootloader'
[6/186] No update step for 'second_stage_bootloader'
[7/186] No patch step for 'second_stage_bootloader'
[8/186] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[9/186] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[10/186] Generating include/generated/driver-validation.h
[11/186] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[12/186] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[13/186] Generating include/generated/offsets.h
[14/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[15/186] Performing configure step for 'second_stage_bootloader'
-- The C compiler identification is GNU 12.1.0
-- The CXX compiler identification is GNU 12.1.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
-- Build files have been written to: /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/bootloader
[16/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32c_sw.c.obj
[17/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[18/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[19/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc32_sw.c.obj
[20/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[21/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[22/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[23/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/posix/getopt/getopt_common.c.obj
[24/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[25/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/fdtable.c.obj
[26/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc16_sw.c.obj
[27/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc8_sw.c.obj
[28/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/crc7_sw.c.obj
[29/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/posix/getopt/getopt.c.obj
[30/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf.c.obj
[31/186] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[32/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[33/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/util/fnmatch/fnmatch.c.obj
[34/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_cache.c.obj
[35/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[36/186] Building C object CMakeFiles/app.dir/src/main.c.obj
[37/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[38/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[39/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/ring_buffer.c.obj
[40/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[41/186] Linking C static library app/libapp.a
[42/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[43/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[44/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg.c.obj
[45/186] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mpsc_pbuf.c.obj
[46/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output.c.obj
[47/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/device_service.c.obj
[48/186] Generating linker_zephyr_pre0.cmd
[49/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_fprintf.c.obj
[50/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_cmds.c.obj
[51/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[52/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/devmem_service.c.obj
[53/186] Generating linker_zephyr_pre1.cmd
[54/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_history.c.obj
[55/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj
[56/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/cpu_idle.S.obj
[57/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_help.c.obj
[58/186] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[59/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_log_backend.c.obj
[60/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/backends/shell_uart.c.obj
[61/186] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[62/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi_on_reset.S.obj
[63/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_mgmt.c.obj
[64/186] Linking C static library zephyr/arch/common/libisr_tables.a
[65/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_cmds.c.obj
[66/186] Linking C static library zephyr/arch/common/libarch__common.a
[67/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_core.c.obj
[68/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_utils.c.obj
[69/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/isr_wrapper.S.obj
[70/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault_s.S.obj
[71/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap_helper.S.obj
[72/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_wildcard.c.obj
[73/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/exc_exit.S.obj
[74/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/prep_c.c.obj
[75/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/irq_manage.c.obj
[76/186] Building C object zephyr/soc/arm/rpi_pico/rp2/CMakeFiles/soc__arm__rpi_pico__rp2.dir/soc.c.obj
[77/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/vector_table.S.obj
[78/186] Building ASM object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/reset.S.obj
[79/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_ops.c.obj
[80/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/nmi.c.obj
[81/186] Generating ../../../include/generated/libc/minimal/strerror_table.h
[82/186] Linking C static library zephyr/soc/arm/rpi_pico/rp2/libsoc__arm__rpi_pico__rp2.a
[83/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/swap.c.obj
[84/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/thread.c.obj
[85/186] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fpu.c.obj
[86/186] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/irq_init.c.obj
[87/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/atoi.c.obj
[88/186] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_page_layout.c.obj
[89/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoul.c.obj
[90/186] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj
[91/186] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/scb.c.obj
[92/186] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/thread_abort.c.obj
[93/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtol.c.obj
[94/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoll.c.obj
[95/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/abort.c.obj
[96/186] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/sys_clock_init.c.obj
[97/186] Building C object zephyr/arch/arch/arm/core/aarch32/cortex_m/CMakeFiles/arch__arm__core__aarch32__cortex_m.dir/fault.c.obj
[98/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/bsearch.c.obj
[99/186] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell.c.obj
[100/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/strtoull.c.obj
[101/186] Building C object zephyr/arch/arch/arm/core/aarch32/CMakeFiles/arch__arm__core__aarch32.dir/fatal.c.obj
[102/186] Linking C static library zephyr/arch/arch/arm/core/aarch32/cortex_m/libarch__arm__core__aarch32__cortex_m.a
[103/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strerror.c.obj
[104/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strncasecmp.c.obj
[105/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strstr.c.obj
[106/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/strspn.c.obj
[107/186] Linking C static library zephyr/arch/arch/arm/core/aarch32/libarch__arm__core__aarch32.a
[108/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/exit.c.obj
[109/186] Linking C static library zephyr/drivers/console/libdrivers__console.a
[110/186] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/cortex_m_systick.c.obj
[111/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/string/string.c.obj
[112/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/qsort.c.obj
[113/186] Linking C static library zephyr/libzephyr.a
[114/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/math/sqrtf.c.obj
[115/186] Building C object zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/uart_rpi_pico.c.obj
[116/186] Linking C static library zephyr/drivers/timer/libdrivers__timer.a
[117/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/sprintf.c.obj
[118/186] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_rpi_pico.c.obj
FAILED: zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_rpi_pico.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DKERNEL -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I../../../../../../include -Izephyr/include/generated -I../../../../../../soc/arm/rpi_pico/rp2 -I../../../../../../lib/posix/getopt/. -I../../../../../../lib/util/fnmatch/. -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_base/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_clocks/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_watchdog/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_xosc/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_pll/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_irq/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_sync/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_timer/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_resets/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/pico_bootrom/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2040/hardware_regs/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2040/hardware_structs/include -I/__w/zephyr/modules/hal/rpi_pico/src/common/pico_base/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/pico_platform/include -I../../../../../../modules/hal_rpi_pico -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_gpio/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_uart/include -I/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/hardware_flash/include -isystem ../../../../../../lib/libc/minimal/include -isystem /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include -isystem /opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.1.0/include-fixed -fno-strict-aliasing -Os -imacros /__w/zephyr/zephyr/twister-out/adafruit_kb2040/samples/drivers/flash_shell/sample.drivers.flash.shell/zephyr/include/generated/autoconf.h -ffreestanding -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m0plus -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.15.2/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-main -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -Werror -fno-asynchronous-unwind-tables -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/drivers/flash_shell=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections -D_POSIX_THREADS -std=c99 -nostdinc -MD -MT zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_rpi_pico.c.obj -MF zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_rpi_pico.c.obj.d -o zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_rpi_pico.c.obj -c /__w/zephyr/zephyr/drivers/flash/flash_rpi_pico.c
In file included from ../../../../../../include/zephyr/device.h:12,
                 from /__w/zephyr/zephyr/drivers/flash/flash_rpi_pico.c:12:
../../../../../../include/zephyr/devicetree.h:2277:32: error: 'DT_CHOSEN_zephyr_flash_controller_REG_IDX_0_VAL_ADDRESS' undeclared here (not in a function); did you mean 'DT_N_S_soc_S_flash_controller_18000000_REG_IDX_0_VAL_ADDRESS'?
 2277 | #define DT_CHOSEN(prop) DT_CAT(DT_CHOSEN_, prop)
      |                                ^~~~~~~~~~
../../../../../../include/zephyr/devicetree.h:3905:33: note: in definition of macro 'DT_CAT4'
 3905 | #define DT_CAT4(a1, a2, a3, a4) a1 ## a2 ## a3 ## a4
      |                                 ^~
../../../../../../include/zephyr/devicetree.h:2086:30: note: in expansion of macro 'DT_REG_ADDR_BY_IDX'
 2086 | #define DT_REG_ADDR(node_id) DT_REG_ADDR_BY_IDX(node_id, 0)
      |                              ^~~~~~~~~~~~~~~~~~
/__w/zephyr/zephyr/drivers/flash/flash_rpi_pico.c:35:26: note: in expansion of macro 'DT_REG_ADDR'
   35 | #define SSI_BASE_ADDRESS DT_REG_ADDR(DT_CHOSEN(zephyr_flash_controller))
      |                          ^~~~~~~~~~~
../../../../../../include/zephyr/devicetree.h:2277:25: note: in expansion of macro 'DT_CAT'
 2277 | #define DT_CHOSEN(prop) DT_CAT(DT_CHOSEN_, prop)
      |                         ^~~~~~
/__w/zephyr/zephyr/drivers/flash/flash_rpi_pico.c:35:38: note: in expansion of macro 'DT_CHOSEN'
   35 | #define SSI_BASE_ADDRESS DT_REG_ADDR(DT_CHOSEN(zephyr_flash_controller))
      |                                      ^~~~~~~~~
/__w/zephyr/zephyr/drivers/flash/flash_rpi_pico.c:61:42: note: in expansion of macro 'SSI_BASE_ADDRESS'
   61 | static ssi_hw_t *const ssi = (ssi_hw_t *)SSI_BASE_ADDRESS;
      |                                          ^~~~~~~~~~~~~~~~
[119/186] Building C object zephyr/drivers/flash/CMakeFiles/drivers__flash.dir/flash_shell.c.obj
[120/186] Linking C static library zephyr/drivers/serial/libdrivers__serial.a
[121/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/math/sqrt.c.obj
[122/186] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj
[123/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdlib/malloc.c.obj
[124/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/fprintf.c.obj
[125/186] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_rpi_pico.c.obj
[126/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/stdout/stdout_console.c.obj
[127/186] Performing build step for 'second_stage_bootloader'
[1/2] Building ASM object CMakeFiles/boot_stage2.dir/__w/zephyr/modules/hal/rpi_pico/src/rp2_common/boot_stage2/boot2_w25q080.S.obj
[2/2] Linking ASM executable boot_stage2
[128/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[129/186] Building C object zephyr/drivers/reset/CMakeFiles/drivers__reset.dir/reset_rpi_pico.c.obj
[130/186] Building C object zephyr/lib/libc/minimal/CMakeFiles/lib__libc__minimal.dir/source/time/gmtime.c.obj
[131/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[132/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[133/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[134/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[135/186] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
ninja: build stopped: subcommand failed.
```

**Environment (please complete the following information):**
 - OS: Ubuntu 20.04
 - Toolchain: Zephyr SDK 0.15.2
 - Commit SHA: https://github.com/zephyrproject-rtos/zephyr/commit/432ff20a72e1bfbc95b40c589b55075d4d82f22e

**Comment by yonsch** [comment]
Is this resolved by #54432 ?

**Comment by stephanosio** [comment]
> Is this resolved by #54432 ?

Yes, I was filing these issues based on the last full build from 3 days ago, I can confirm that it passes now.
